modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -325,24 +445,57 @@ impl<T: Config<I>, I: 'static> SendXcm for Pallet<T, I> {
 			return Err(SendError::ExceedsMaxMessageSize)
 		}
 
+		// compute bridge id from universal locations
+		let bridge_destination_relative_location =
+			dest.as_ref().map(|dest| dest.clone()).ok_or(SendError::MissingArgument)?;
+		let bridge_origin_universal_location = T::UniversalLocation::get();
+		let bridge_destination_universal_location: InteriorMultiLocation =
+			bridge_origin_universal_location
+				.into_location()
+				.appended_with(bridge_destination_relative_location)
+				.map_err(|_| SendError::NotApplicable)?
+				.try_into()
+				.map_err(|_| SendError::NotApplicable)?;
+		let bridge_id = bridge_id_from_locations(
+			&T::UniversalLocation::get(),
+			&bridge_destination_universal_location,
+		);
+
 		// just use exporter to validate destination and insert instructions to pay message fee
 		// at the sibling/child bridge hub
 		//
 		// the cost will include both cost of: (1) to-sibling bridg hub delivery (returned by
 		// the `Config::ToBridgeHubSender`) and (2) to-bridged bridge hub delivery (returned by
 		// `Self::exporter_for`)
 		ViaBridgeHubExporter::<T, I>::validate(dest, xcm)
-			.map(|(ticket, cost)| ((message_size, ticket), cost))
+			.map(|(ticket, cost)| ((bridge_id, message_size, ticket), cost))
 	}
 
 	fn deliver(ticket: Self::Ticket) -> Result<XcmHash, SendError> {
 		// use router to enqueue message to the sibling/child bridge hub. This also should handle
 		// payment for passing through this queue.
-		let (message_size, ticket) = ticket;
-		let xcm_hash = ViaBridgeHubExporter::<T, I>::deliver(ticket)?;
+		let (bridge_id, message_size, ticket) = ticket;
+		let mut bridge =
+			Bridges::<T, I>::get(bridge_id).ok_or_else(|| SendError::Transport("UnknownBridge"))?;
+		let xcm_hash = if bridge.is_suspended() {
+			// TODO: this is the most worrysome code here - I'm not sure that it is ok to encode and
+			// save the **ticket**. Maybe we need to support just some ticket portion and then
+			// reconstruct the ticket?
+			let encoded_ticket = ticket.encode();
+			// TODO: not sure if there's some consistency for computing that?
```

Comment:
I've seen how it is computed in other senders (`blake2_256`), but haven't seen how it is actually used. Shall we care about this value at all?